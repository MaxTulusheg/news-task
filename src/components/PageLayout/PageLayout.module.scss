@import '~/src/styles';

.wrapper {
  display: grid;
  gap: 1rem;
  grid-template-columns: 2fr 1fr;
  max-width: 100%;
  margin: 0 auto;
  padding: 1rem;

  @include tablet {
    max-width: 696px;
  }

  @include desktop {
    max-width: 960px;
  }

  @include largeDesktop {
    max-width: 1120px;
  }
}

.header {
  grid-column: 1 / 3;
}

.content {
  grid-column: 1 / 3;
  
  @include tablet {
    grid-column: 1 / 2;
  }
}

.aside {
  grid-column: 1 / 3;
  grid-row: 1;
  align-self: start;
  
  @include tablet {
    position: sticky;
    top: 65px;
    grid-column: 2 / 3;
  }
}
