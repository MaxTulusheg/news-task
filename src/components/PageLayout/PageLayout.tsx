import React, { FC, PropsWithChildren } from 'react';

import Header from 'components/Header/Header';
import HotNewsWidget from 'components/HotNewsWidget/HotNewsWidget';

import styles from './PageLayout.module.scss';

const PageLayout: FC<PropsWithChildren> = ({ children }) => (
  <>
    <Header className={styles.header} />
    <div className={styles.wrapper}>
      <main className={styles.content}>{children}</main>
      <HotNewsWidget className={styles.aside} />
    </div>
  </>
);

export default PageLayout;
