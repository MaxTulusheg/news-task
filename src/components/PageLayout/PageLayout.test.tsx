import React from 'react';
import { render } from '@testing-library/react';

import PageLayout from './PageLayout';

describe('PageLayout component', () => {
  it('renders correct structure', () => {
    const { container } = render(<PageLayout>test</PageLayout>);

    expect(container).toMatchSnapshot();
  });
});
