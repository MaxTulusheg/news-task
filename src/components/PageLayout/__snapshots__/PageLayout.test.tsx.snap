// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout component renders correct structure 1`] = `
<div>
  <header
    class="wrapper header"
  >
    <nav
      class="navigation"
    >
      <a
        class="link"
        href="/"
        title="News"
      >
        <img
          alt="News logo"
          class="logo"
          src="logo.svg"
        />
      </a>
    </nav>
  </header>
  <div
    class="wrapper"
  >
    <main
      class="content"
    >
      test
    </main>
    <aside
      class="wrapper aside"
    >
      <h3
        class="title heading h3"
      >
        <img
          alt="Icon"
          class="icon icon"
          src="fire.svg"
        />
        <a
          class="title"
          href="#hot-news"
          title="See all hot news"
        >
          Hot News
        </a>
        <img
          alt="Icon"
          class="icon icon"
          src="fire.svg"
        />
      </h3>
      <div
        class="wrapper loader"
        data-testid="loader"
      >
        <div
          class="loader loader"
        />
      </div>
      <button
        class="refresh"
        disabled=""
        title="Refresh hot news"
      >
        Refresh
        <img
          alt="Icon"
          class="icon refreshIcon refreshLoad"
          src="refresh.svg"
        />
      </button>
    </aside>
  </div>
</div>
`;
